framework/src/userobject/SolutionUserObject.C

```diff
@@ -645,6 +645,8 @@ SolutionUserObject::evalMeshFunction(const Point & p, std::string var_name, unsi
 
   // Extract the variable index for the MeshFunction(s), must use iterator b/c of const
   std::map<std::string, unsigned int>::const_iterator it = _local_variable_index.find(var_name);
+  if (it == _local_variable_index.end())
```

Comment:
The SolutionFunction calls this method and passes in the name of the variable for which it's asking for a value. The SolutionUserObject doesn't know what the name of the variable that will be passed in is in its constructor. I agree that it's preferable to catch this earlier. Maybe the thing to do is to have a method on SolutionUserObject that returns an index to the variable. That would get called by SolutionFunction early on and get stashed away, and then we would pass in that index instead of the variable name in these methods that give the value. That would be more efficient as well.